src/core/injections/controller/checks.py

```diff
@@ -588,14 +588,9 @@ def ignore_google_analytics_cookie(cookie):
 """
 def newline_fixation(payload):
   payload = _urllib.parse.unquote(payload)
-  if settings.END_LINE.CR in payload:
-    #_ = payload.find("\r\n") + 1
-    #payload = _urllib.parse.quote(payload[:_]) + payload[_:]
-    payload = payload.replace(settings.END_LINE.CR,"%0d")
-  if settings.END_LINE.LF in payload:
-    #_ = payload.find("\n") + 1
-    #payload = _urllib.parse.quote(payload[:_]) + payload[_:]
-    payload = payload.replace(settings.END_LINE.LF,"%0a")
+  payload = _urllib.parse.quote(payload, safe='')
+  payload = payload.replace(settings.END_LINE.CR, "%0D%0A" if '\r\n' in payload else "%0D")
+  payload = payload.replace(settings.END_LINE.LF, "%0A")
```

Comment:
Considerations:

- The new version assumes it's safe to fully percent-encode the payload. If the payload contains already encoded sequences, double encoding could occur (e.g., `%25` for `%`). You might need to ensure `unquote()` is thorough enough.

- The detection of `\r\n` after quoting might miss cases where encoding already changed the string - consider checking for `\r\n` before quoting.